servant-client/src/Servant/Common/Req.hs

```diff
@@ -214,6 +215,10 @@ instance MonadBaseControl IO ClientM where
   -- restoreM :: StM ClientM a -> ClientM a
   restoreM st = ClientM (restoreM st)
 
+-- | Try clients in order, last error is preserved.
+instance Semigroup (ClientM a) where
+  a <> b = a `catchError` \_ -> b
```

Comment:
I think using `<>` for `<!>` is morally wrong. The appropriate instance here would be:

```haskell
-- this requires semigroupoids
instance Alt ClientM where
  a <!> b = a `catchError` \_ -> b

-- this is probably useless in practice, so I wouldn't even add this instance
instance Semigroup a => Semigroup (ClientM a) where
  (<>) = liftA2 (<>)
```

Instead of making a morally wrong `Semigroup` instance I would rather add `semigroupoids` dep.
I can't tell right now, but I suspect that it is already a transitive dependency in more than one way.